Sources/Logging.php

```diff
@@ -17,6 +17,29 @@
 	die('No direct access...');
 
 /**
+ * Truncate the GET array to a specified length
+ * @param array $arr The array to truncate
+ * @param max_length $max_length The upperbound on the length
+ *
+ * @return array The truncated array
+ */
+function truncateArray($arr, $max_length=2048) {
```

Comment:
If the maximum length for the database field is 2048 characters, the value of `$max_length` here should be less so that there will always be enough room for the user agent string, etc.